src/tribler/core/utilities/utilities.py

```diff
@@ -107,10 +107,10 @@ def parse_magnetlink(url):
 
             elif key == "xt" and value.startswith("urn:btih:"):
                 # vliegendhart: Adding support for base32 in magnet links (BEP 0009)
-                encoded_infohash = value[9:49]
+                encoded_infohash = value[9:]
                 if len(encoded_infohash) == 32:
                     xt = b32decode(encoded_infohash.upper())
-                else:
+                elif len(encoded_infohash) == 40:
                     xt = binascii.unhexlify(encoded_infohash)
```

Comment:
If we expect that the infohash value can be invalid, we also should guard `b32decode(...)` and `binascii.unhexlify(...)` calls with try/except blocks, and return the `None` infohash value in case of exception